sql/cell_value.cc

```diff
@@ -199,14 +199,22 @@ CellValue
 CellValue::
 parse(const char * s_, size_t len, StringCharacteristics characteristics)
 {
-    std::string cstring(s_, len); // ensure the buffer is null terminated
-    const char * s = cstring.c_str();
+    static constexpr size_t NUMERICAL_BUFFER = 64;
+
+    // if the string is longer than 64 characters it can't realistically
+    // be a numerical value
+    if (len > NUMERICAL_BUFFER)
+        return CellValue(s_, len, characteristics);
 
     if (len == 0)
         return CellValue();
 
+    // this ensures that our buffer is null terminated as required below
+    char s[NUMERICAL_BUFFER + 1] = {0};
```

Comment:
Instead of initializing to zero then memcpy, do the memcpy and then set the final byte to zero